### DISCUSSION TITLE
Fish custom key binding with `y` function doesn't update fish prompt directory

### DISCUSSION THREAD

**Discussion by jcelaya775** [sxyazi/yazi#1964] [1. Q&A]
### What system are you running Yazi on?

macOS

### What terminal are you running Yazi in?

kitty 0.34.1

### `yazi --debug` output

```Shell
Yazi
    Version: 0.3.3 (Homebrew 2024-09-04)
    Debug  : false
    OS     : macos-aarch64 (unix)

Ya
    Version: 0.3.3 (Homebrew 2024-09-04)

Emulator
    Emulator.via_env: ("tmux-256color", "tmux")
    Emulator.via_csi: Ok(Kitty)
    Emulator.detect : Kitty

Adapter
    Adapter.matches: Kitty

Desktop
    XDG_SESSION_TYPE           : None
    WAYLAND_DISPLAY            : None
    DISPLAY                    : None
    SWAYSOCK                   : None
    HYPRLAND_INSTANCE_SIGNATURE: None
    WAYFIRE_SOCKET             : None

SSH
    shared.in_ssh_connection: false

WSL
    WSL: false

Variables
    SHELL              : Some("/opt/homebrew/bin/fish")
    EDITOR             : Some("nvim")
    VISUAL             : None
    YAZI_FILE_ONE      : None
    YAZI_CONFIG_HOME   : Some("/Users/jorge/.config/yazi/")

Text Opener
    default: Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })
    block  : Some(Opener { run: "${EDITOR:-vi} \"$@\"", block: true, orphan: false, desc: "$EDITOR", for_: None, spread: true })

Multiplexers
    TMUX               : true
    tmux version       : tmux 3.5a
    ZELLIJ_SESSION_NAME: None
    Zellij version     : No such file or directory (os error 2)

Dependencies
    file             : 5.41
    ueberzugpp       : No such file or directory (os error 2)
    ffmpegthumbnailer: 2.2.3
    magick           : 7.1.1-41
    fzf              : 0.56.3
    fd               : 10.2.0
    rg               : 14.1.1
    chafa            : No such file or directory (os error 2)
    zoxide           : 0.9.6
    7z               : No such file or directory (os error 2)
    7zz              : 24.08
    jq               : 1.7.1


--------------------------------------------------
When reporting a bug, please also upload the `yazi.log` log file - only upload the most recent content by time.
You can find it in the "/Users/jorge/.local/state/yazi" directory.
```

### Describe the bug

In fish shell, I configured a custom key binding (`ctrl+y`) to run the [shell wrapper function](https://yazi-rs.github.io/docs/quick-start#shell-wrapper), but when I press `ctrl+y` in fish and then switch directories inside yazi and then press `q`, the fish prompt is still showing the original directory unless I enter another command, and only then does it show the correct directory that yazi `cd`'d into. The is probably a fish prompt issue, but figured I would bring this up since someone else might be trying the same thing.

### Minimal reproducer

Paste the following into `~/.config/fish/config.fish`:
```fish
function y
      set tmp (mktemp -t "yazi-cwd.XXXXXX")
      yazi $argv --cwd-file="$tmp"
      if set cwd (command cat -- "$tmp"); and [ -n "$cwd" ]; and [ "$cwd" != "$PWD" ]
        builtin cd -- "$cwd"
      end
      rm -f -- "$tmp"
end

bind -M insert \cy 'y'
```

Follow these steps using fish shell:

1. `pwd` (original directory, before pressing `ctrl+y`):
![image](https://github.com/user-attachments/assets/29ce0af8-02d6-492b-91ee-0ee10a61d95f)

2. Press `ctrl+y`:
![image](https://github.com/user-attachments/assets/8a9c9aa1-d4da-4665-a724-f6eeb9ed1b5b)

3. Move into another directory:
![image](https://github.com/user-attachments/assets/07ba515a-b19f-4f41-9cb0-dd9850392105)

4. Press `q` (the correct directory is not being displayed in the fish prompt):
![image](https://github.com/user-attachments/assets/8d3a45de-d659-4f1b-80a6-fb7f8f0536c7)

5. `pwd` again (the fish prompt only shows the correct directory after entering another command):
![image](https://github.com/user-attachments/assets/70e24e47-2829-4d33-9b61-4f9cd06ed42c)

### Anything else?

This behavior doesn't occur when I run `y` in fish (without the custom key binding), and it works as expected. 

### Validations

- [x] I tried the [latest nightly build](https://yazi-rs.github.io/docs/installation#official-binaries), and the issue is still reproducible
- [X] I updated the debug information (`yazi --debug`) input box to the nightly that I tried
- [X] I can reproduce it after disabling all custom configs/plugins (`mv ~/.config/yazi ~/.config/yazi-backup`)

**Comment by sxyazi** [comment]
Does it work if you use `'y\n'` instead of just `'y'`? I have `bindkey -s "^y" "y\n"` for my Zsh and works well, but I'm not sure if it applies to fish as well

**Reply by jcelaya775** [reply]
That specific binding didn't work for me, but I tried something similar for tmux that works. It seems like `y` doesn't "enter" by itself when using a key binding, so this resolves it (while inside tmux). Thanks for the suggestion!

`/Users/jorge/.config/fish`:
```
bind -M insert \cy 'y && tmux send-keys Enter'
```
